ssl-config/src/main/scala/com/avast/sst/ssl/SslContextModule.scala

```diff
@@ -0,0 +1,26 @@
+package com.avast.sst.ssl
+
+import cats.effect.Sync
+import com.typesafe.config.Config
+import com.typesafe.sslconfig.ssl.{
+  ConfigSSLContextBuilder,
+  DefaultKeyManagerFactoryWrapper,
+  DefaultTrustManagerFactoryWrapper,
+  SSLConfigFactory
+}
+import javax.net.ssl.{KeyManagerFactory, SSLContext, TrustManagerFactory}
+
+import scala.language.higherKinds
+
+object SslContextModule {
+
+  /** Initializes [[javax.net.ssl.SSLContext]] from the provided config. */
+  def make[F[_]: Sync](config: Config): F[SSLContext] = Sync[F].delay {
```

Comment:
now, this might come across as way too pedantic, but: I like that we don't use `com.typesafe.config.Config` in our "API". And I would prefer if it stayed that way.

The fact that we're creating the `SSLContext` using `com.typesafe.sslconfig.ssl` is an implementation detail, which be leaking out.

suggestion: create a config case class (as everywhere else) to configure SSLContext and change `make` method to use that

Reply:
Personally we've discussed this and I decided to keep the name because it reflects the dependency used.